packages/core/src/types.ts

```diff
@@ -828,7 +828,11 @@ export interface StateMachine<
   ): StateMachine<TContext, TStateSchema, TEvent, TTypestate>;
 }
 
-export type StateFrom<T> = T extends StateMachine<any, any, any>
+export type StateFrom<
+  T extends
```

Comment:
Is the idea to add to this `extends` whenever we add a new thing that `StateFrom` can extract a state from?

Reply:
Yes, that will work.

Reply:
I've removed this on my branch - https://github.com/mattpocock/xstate/pull/1/files . I don't have super strong opinion about this and definitely having those constraints in place could improve where type errors are reported and so on. We didn't have those in other *From utilities - so I've just gone for the unification for the time being. We can always unify this other way around - although keeping the list of those things is a little bit tedious 🤷‍♂️